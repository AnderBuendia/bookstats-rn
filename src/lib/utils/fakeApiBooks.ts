import { BookStatus } from 'enums/book-status.enum';
import type { Book } from '@Interfaces/domain/book.interface';

const date = new Date('2022');

export const HomeBooks: Book[] = [
  {
    id: '1',
    title: 'The Brothers Karamazov',
    author: 'Dostoievsky',
    status: BookStatus.COMPLETED,
    rating: 80,
    pages: 0,
    userId: 'user1',
    image: null,
    review: null,
    read_pages: [],
    createdAt: date,
    updatedAt: date,
  },
  {
    id: '2',
    title: 'Steppenwolf',
    author: 'Hesse',
    status: BookStatus.READING,
    rating: 40,
    pages: 20,
    userId: 'user2',
    image: null,
    review: null,
    read_pages: [],
    createdAt: date,
    updatedAt: date,
  },
  {
    id: '3',
    title: 'One hundred years of solitude',
    author: 'MÃ¡rquez',
    status: BookStatus.READY,
    rating: 20,
    pages: 417,
    userId: 'user3',
    image: null,
    review: null,
    read_pages: [],
    createdAt: date,
    updatedAt: date,
  },
  {
    id: '4',
    title: 'The Symposium',
    author: 'Plato',
    status: BookStatus.TO_READ,
    rating: 0,
    pages: 111,
    userId: 'user4',
    image: null,
    review: null,
    read_pages: [],
    createdAt: date,
    updatedAt: date,
  },
];
