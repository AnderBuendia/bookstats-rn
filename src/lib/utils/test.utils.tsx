import type { ReactElement } from 'react';
import { render } from '@testing-library/react-native';
import { AppStoreContext } from '@Lib/context/app-store.context';
import AuthStoreContext from 'lib/context/auth-store.context';
import { UIState } from '@Enums/config/ui-state.enum';
import type { ErrorState } from '@Interfaces/config/errors.interface';

export interface RenderOptions {
  providerProps?: {
    uiState?: UIState | null;
    errorState?: ErrorState;
    setUiState?: () => void;
    setErrorState?: () => void;
  };
  renderOptions?: {};
}

const customRender = (ui: ReactElement, options: RenderOptions = {}) => {
  const authProviderValues = {
    uiState: options.providerProps?.uiState,
    setUiState: options.providerProps?.setUiState,
  };

  const appProviderValues = {
    errorState: options.providerProps?.errorState,
    setErrorState: options.providerProps?.setErrorState,
  };

  return render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => (
      <AuthStoreContext.Provider value={authProviderValues}>
        <AppStoreContext.Provider value={appProviderValues}>
          {children}
        </AppStoreContext.Provider>
      </AuthStoreContext.Provider>
    ),
    ...options.renderOptions,
  });
};

export * from '@testing-library/react-native';
// override render export
export { customRender as render };
