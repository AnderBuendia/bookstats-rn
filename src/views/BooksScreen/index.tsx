import type { FC } from 'react';
import { View, Text } from '@Components/generic/Theme/Themed';

export interface BooksScreenProps {}

const BooksScreen: FC<BooksScreenProps> = () => {
  return (
    <View>
      <Text>Hello from BooksScreen</Text>
    </View>
  );
};

export default BooksScreen;
