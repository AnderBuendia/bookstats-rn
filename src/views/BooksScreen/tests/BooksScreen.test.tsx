import { Button, Text, TextInput, View } from 'react-native';
import { render, fireEvent, waitFor } from '@Lib/utils/test.utils';
import BooksScreen from '@Views/BooksScreen';

const createTestProps = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

describe('Test Auth Screen', () => {
  let props: any;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('Should show Books title text', () => {
    render(<BooksScreen {...props} />);

    expect(/Bookstats/i).toMatchSnapshot();
  });
});
