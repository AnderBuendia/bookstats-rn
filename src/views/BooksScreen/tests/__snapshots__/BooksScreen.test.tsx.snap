// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Auth Screen Should show Books title text 1`] = `/Bookstats/i`;
